src/certs.rs

```
@@ -196,23 +205,18 @@ pub fn get_ext_certs(args: CertificatesArgs) -> Result<()> {
         .get_ext_report(None, Some(request_data), None)
         .context("Failed to get extended report.")?;
 
-    // Check if the returned table is empty
-    if certificates.is_empty() {
-        return Err(anyhow::anyhow!(
-            "The certificate chain is empty! Certificates probably not loaded by the host."
-        ));
-    }
-
     // Create certificate directory if missing
     if !args.certs_dir.exists() {
         fs::create_dir(args.certs_dir.clone()).context("Could not create certs folder")?;
     };
 
-    // Write certs into directory
-    for cert in certificates.iter() {
-        let path = args.certs_dir.clone();
+    // If certificates are present, write certs into directory
+    if let Some(ref certificates) = certificates {
+        for cert in certificates.iter() {
+            let path = args.certs_dir.clone();
 
-        write_cert(path, &cert.cert_type, &cert.data, args.encoding)?;
+            write_cert(path, &cert.cert_type, &cert.data, args.encoding)?;
+        }
```

Do we want to return an error if certificate is empty?